molo/core/templatetags/core_tags.py

```diff
@@ -3,22 +3,43 @@
 from django.utils.safestring import mark_safe
 from markdown import markdown
 
+from wagtail.wagtailcore.models import Site
+from wagtail.contrib.settings.context_processors import SettingsProxy
+
 from molo.core.models import Page, SiteLanguage, ArticlePage, SectionPage
 
 register = template.Library()
 
 
+def get_pages(qs, locale):
+    language = SiteLanguage.objects.filter(locale=locale).first()
+    site = Site.objects.get(is_default_site=True)
+    settings = SettingsProxy(site)
```

Comment:
Instead of using the proxy here, can you use `SiteSettings.for_site(request.site)` and just pass the request into this function.